File: core/src/frontendServer/server/models/Weapon.js
class Weapon {

    constructor(reloadTime, bulletsInChamber, magazineRefillTime, projectileType) {
        this.reloadTime = reloadTime;
        this.lastShoot = 0;
        this.maxBulletsInChamber = bulletsInChamber;
        this.bulletsInChamber = bulletsInChamber;
        this.magazineRefillTime = magazineRefillTime;
        this.projectileType = projectileType;
    }

    canShoot() {
        return new Date().getTime() - this.lastShoot > this.reloadTime & this.bulletsInChamber > 0
    }

    shoot() {
        if (this.canShoot()) {
            this.lastShoot = new Date().getTime();
            this.bulletsInChamber--;
            return true
        } else return false
    }

    reload() {
        this.bulletsInChamber = this.maxBulletsInChamber
    }
}

module.exports = Weapon;

File: core/src/com/mygdx/game/model/weapon/Weapon.kt
package com.mygdx.game.model.weapon

import com.badlogic.gdx.utils.TimeUtils

abstract class Weapon(private val reloadTime: Long, val maxBulletsInChamber: Int, val type: String) {
    private var lastShoot = 0L

    var bulletsInChamber = maxBulletsInChamber

    fun canShoot() = TimeUtils.millis() - lastShoot > reloadTime

    fun shoot() = if (canShoot()) {
        lastShoot = TimeUtils.millis()
        true
    } else false

}